### ISSUE TITLE
Add ability to use stdout and stdin instead of /dev/tty

### ISSUE THREAD

**Issue by tmcguire** [gyscos/cursive#456]
* Report type: Suggestion
* Operating system: Ubuntu 19.10
* Backend used: ncurses (default)
* Cursive version: crates.io 0.14.1

# Problem description

When debugging with [vscode-lldb](https://github.com/vadimcn/vscode-lldb), `/dev/tty` is not available due to the way lldb spawns the debugee. See https://github.com/vadimcn/vscode-lldb/issues/296 for the corresponding ticket.

Since I'd like to be able to launch & debug my cursive application from within vscode, a workaround for that could be to open `stdin` and `stdout` instead of the unavailable `/dev/tty`.

Note that I haven't actually tried whether this will work and fix the problem, so my assumption might be incorrect.

**Comment by gyscos** [comment]
Hi, and thanks for the report!

I think it should be possible to make backends configurable, so you could create one pointing to stdin/stdout.

(Note that some backends may behave strangely without a tty; not sure how they would fetch the window size and a few other details.)

**Comment by jarontai** [comment]
I second this feature. I have to switch to old version if I want to debug in vscode.

**Comment by gyscos** [comment]
In the meantime, have you tried running the application separately and attaching the vscode debugger to the process?

**Comment by jarontai** [comment]
> In the meantime, have you tried running the application separately and attaching the vscode debugger to the process?

I tried the attach approach with version 0.14.0 and that works for me, thank you.

**Comment by tmcguire** [comment]
> In the meantime, have you tried running the application separately and attaching the vscode debugger to the process?

Yes, that works for me and I'm using that as a workaround for now.

**Comment by gyscos** [comment]
The latest commit brings some experimental function to initialize a `ncurses` or `termion` backend using a different file.

For example:

```rust
let mut siv = cursive::Cursive::try_new(|| cursive::backends::termion::init_stdio()).unwrap();
```

A similar method exists for `cursive::backends::curses::n::Backend`.

There's also `init_with_files` if you want to manually specify the input/output files. For now, these have to be `File` for termion, and valid file paths for ncurses. Termion might be updated to use any `Write`/`Read`, but it's not clear that's possible for ncurses.

**Comment by Rushmore75** [comment]
Did you ever get it working normally with vscode?

**Comment by mikem-zed** [comment]
I opened https://github.com/gyscos/cursive/pull/716 to address the same problem with crossterm. Only stdout can be set for now
